app/controllers/posts_controller.rb

```diff
@@ -1,17 +1,16 @@
 class PostsController < ApplicationController
   def index
     user_id = params[:user_id]
-    @user = User.find(user_id)
-    @posts = Post.where(author_id: user_id)
+    @user = User.includes(posts: { comments: [:author] }).find(user_id)
+    @posts = @user.posts
```

Comment:
## [OPTIONAL]

![image](https://user-images.githubusercontent.com/65068771/217387950-b55d6339-ea25-49e6-b171-43b7be8e85fa.png)

- [ ] Nice job on adding screenschorts before and after solving n+1 problems, However some of your test ar not running. I think the reason should be the line of code I have selected above. In your `posts/show.html.erb` view you are accessing a single post with `@post` but in your controller there is no where you are getting a single @post rather you are geting all @posts. You can fix it by adding a single post in your controller `@post = @user.posts.find(params[:id])`

- [ ] Also I think it would be better if you fix the error in your `views/posts/show.html.rb` that requires also to set the @comment in your controller `@comments = @post.comments`